### PR TITLE
[13.0][ADD]hr_employee_leave_quota

### PR DESCRIPTION
**Pull request by watcharaporn-git**
This module allows Employees can see the number of days they can take leave,
such as sick leave, leave, and can also submit a leave through the system.
Which the manager can confirm the leave

@kittiu can you reviews please?

### CODE REVIEW

**Comment by kittiu** [comment]
Firstly, please see travis error messages. Please do pre-commit.

File: hr_employee_leave_quota/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
+{
+    "name": "HR Employee Leave Quota",
+    "version": "13.0.1.0.0",
+    "author": "Brainbean Apps, Odoo Community Association (OCA)",
```
Comment (kittiu) [inline hr_employee_leave_quota/__manifest__.py]:
```suggestion
    "author": "Ecosoft,Odoo Community Association (OCA)",
```

File: hr_employee_leave_quota/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "license": "AGPL-3",
+    "installable": True,
+    "application": False,
+    "depends": ["hr"],
+    "data": [
+        "security/ir.model.access.csv",
+        "views/hr_employee_leave_quota_veiws.xml",
+
```
Comment (kittiu) [inline hr_employee_leave_quota/__manifest__.py]:
```suggestion
``

File: hr_employee_leave_quota/models/hr_employee_leave_quota.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
+
```
Comment (kittiu) [inline hr_employee_leave_quota/models/hr_employee_leave_quota.py:4]: In all .py files, add this,
```
# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
```

File: hr_employee_leave_quota/models/hr_employee_leave_quota.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    leave_type_ids = fields.One2many(
+        string="Holiday Quota",
+        comodel_name="hr.employee.leave.quota",
+        inverse_name="employee_id",
+    )
+    cancel_reason_id = fields.Many2one(
+        'leave.quota.cancel.reason',
+        string="Reason for cancellation",
+        readonly=True,
+        ondelete="restrict")
```
Comment (kittiu) [inline hr_employee_leave_quota/models/hr_employee_leave_quota.py]: Do we need this yet?
```suggestion
```

File: hr_employee_leave_quota/models/hr_employee_leave_quota.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    used_days = fields.Integer(
+        string="Used (days)",
+    )
+    balance = fields.Integer(
+        string="Balance",
+        onchange="_onchange_balance",
+    )
+
+    @api.onchange('quota', 'used_days')
+    def _onchange_balance(self):
```
Comment (kittiu) [inline hr_employee_leave_quota/models/hr_employee_leave_quota.py]: This should be compute field, not onchange, right?

File: hr_employee_leave_quota/models/hr_employee_leave_quota.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+        string="Approve By",
+        readonly=True,
+    )
+    request_line_ids = fields.One2many(
+        'hr.employee.leave.request.line',
+        inverse_name="request_id",
+    )
+    state = fields.Selection(
+        [("draft", "Draft"), ("waiting approval", "Waiting Approval"),
+            ("approved", "Approved"), ("rejected", "Rejected")],
```
Comment (kittiu) [inline hr_employee_leave_quota/models/hr_employee_leave_quota.py]:
```suggestion
        [("draft", "Draft"),
         ("waiting", "Waiting Approval"),
         ("approved", "Approved"),
         ("rejected", "Rejected")],
```

File: hr_employee_leave_quota/models/hr_employee_leave_quota.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+        readonly=True,
+        index=True,
+        copy=False,
+        default="draft",
+        track_visibility="onchange",
+    )
+
+    def _default_hr_employee(self):
+        Employee = self.env["hr.employee"]
+        employee_id = Employee.search([("user_id", "=", self.env.user.id)])
```
Comment (kittiu) [inline hr_employee_leave_quota/models/hr_employee_leave_quota.py]:
```suggestion
        employee = Employee.search([("user_id", "=", self.env.user.id)])
```
If the variable is object, try not to end with "_id".

File: hr_employee_leave_quota/models/hr_employee_leave_quota.py [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+        return employee_id.id
+
+    @api.model
```
Comment (kittiu) [inline hr_employee_leave_quota/models/hr_employee_leave_quota.py]:
```suggestion
```

File: hr_employee_leave_quota/models/hr_employee_leave_quota.py [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+        self.write({"state": "draft"})
+
+    def button_submit_approved(self):
+        self.write({"state": "waiting approval"})
+
+    def button_approve(self):
+        for rec in self:
+            Employee = rec.employee_id
+            Quota = Employee.leave_type_ids
+            Resuest_line = rec.request_line_ids
```
Comment (kittiu) [inline hr_employee_leave_quota/models/hr_employee_leave_quota.py:97]:
```suggestion
            employee = rec.employee_id
            leave_types = Employee.leave_type_ids
            request_lines = rec.request_line_ids
```

**Comment by jarroyomorales** [comment]
Doesnt this overlap with core `hr_holidays` module?

File: hr_employee_leave_quota/models/hr_employee_leave_quota.py [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+        string="Date End",
+    )
+    leave_days = fields.Float(
+        string="Leave Days",
+        compute="_compute_leave_days",
+        inverse="_inverse_leave_days",
+    )
+    balance = fields.Integer(
+        string="Balance",
+        onchange="_onchange_balance"
```
Comment (kittiu) [inline hr_employee_leave_quota/models/hr_employee_leave_quota.py]: There is no such thing as onchange here.

**Comment by kittiu** [comment]
> Doesnt this overlap with core `hr_holidays` module?

@jarroyomorales sorry, this is my attempt to explain how to work with OCA to our new staff.
Note: we will move it out soon, sorry for confusion.

**Comment by jarroyomorales** [comment]
@kittiu Oh okay! No problem then

**Comment by Saran440** [comment]
Closed.

**Pull request closed without merging**